src/devices/sound/tms5220.cpp

```diff
@@ -1972,13 +1962,15 @@ void tms5220_device::write_data(uint8_t data)
 uint8_t tms5220_device::read_status()
 {
 	// prevent debugger from changing the internal state
-	if (machine().side_effects_disabled()) return 0;
+	if (!machine().side_effects_disabled())
+		m_stream->update(); /* bring up to date first */
 
 	if (!m_true_timing)
 	{
-		/* bring up to date first */
-		m_stream->update();
-		return status_read();
+		// prevent debugger from changing the internal state
+		if (machine().side_effects_disabled())
+			return status_read(false);
+		return status_read(true);
```

Comment:
You could just do `return status_read(!machine().side_effects_disabled());` here.